src/libs/WebsocketProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import { createWS } from "@solid-primitives/websocket";
+import { createSignal } from "solid-js";
+
+import { WebsocketContext } from "./websocket";
+
+export default function WebsocketProvider(props: any) {
+  const websocket = createWS("ws://localhost:3000/ws");
+  const [messages, setMessages] = createSignal<Message[]>([]);
+
+  // TODO: Don't do message logic inside the websocket provider.
```

Comment:
Given that websockets are needed to create messages, I think it makes sense to leave them within this provider. A name change may make this feel more natural, what are your thoughts?

Reply:
I actually thought of this, and it may be something to consider for the reasons you mentioned. Since this is wrapping the entire router, we could make it more "global" in a sense, not using it only for websocket related stuff.

Reply:
Then again, I don't know where stores come into play in all of this...

Reply:
If you've never used something like [Jotai](https://github.com/pmndrs/jotai)/[Zustand](https://github.com/pmndrs/zustand), stores act like context in functional programming where you store feature-/system-relevant state, then pass it to everything that needs it.

Stores are similar except rather than context localising state and passing it around, stores _globalise_ state so that it can be accessed from anywhere that needs it, while centralising everything in 1 place - in our case this would be the module that we define then import whenever we need to do some form of message manipulation.